support-lib/jni/djinni_support.cpp

```diff
@@ -22,6 +22,7 @@
 #include <cstring>
 
 static_assert(sizeof(jlong) >= sizeof(void*), "must be able to fit a void* into a jlong");
+static_assert(sizeof(wchar_t) == 2 || sizeof(wchar_t) == 4, "wchar_t must be represented by UTF-16 or UTF-32 encoding");
```

Comment:
I think it would be better if this static_assert didn't trigger if wstrings aren't in use, since they're an optional feature.  I think you could instead put a static_assert inside of the default implementation of your template functions, so it would only trigger if that function is called, and thus instantiated.